tests/Unit/Repository/GroupRepositoryTest.php

```diff
@@ -48,7 +85,32 @@ protected function tearDown(): void
 
     public function testFindAllFor(): void
     {
-        /* @todo This test is incomplete. */
-        self::markTestIncomplete();
+        $registration = $this->em->getRepository(LocalAccount::class)->findAll()[0];
+        $groups = $this->em->getRepository(Group::class)->findAllFor($registration);
+
+        self::assertTrue(count($groups) > 0);
+    }
+
+    public function testFindSubGroupsFor(): void
+    {
+        $registration = $this->em->getRepository(LocalAccount::class)->findAll()[0];
+        $group = $this->em->getRepository(Group::class)->findAllFor($registration)[0];
+
+        $allgroups = $this->em
+            ->getRepository(Group::class)
+            ->findSubGroupsFor($group);
+
+        self::assertTrue(count($allgroups) > 0);
```

Comment:
It depends on a test to test bases whether or not there is any use in a more precise test. 
In most cases all you can additionally test is whether all the found results also hold the original property that you are looking for. But its already a given that it will, otherwise it will return 0 results. This Count() > 0 is something that we also done in most other repository tests. But if u deem it necessary I can implement a property check for the RepositoryTests

Reply:
I feel like there is more room to make the asserts more specific, other than checking whether any result is returned. For example with this function, the property that needs to hold is that each group from $allgroups should include $group somewhere in its parent hierarchy. This means that line 103 could be replaced with:
```php
foreach ($allgroups as $parent) {
    // keep iterating over parents until $group is found
    while (true) {
        $parent = $parent->getParent();
        if ($parent == $group) {
            // $parent is in hierarchy, success! Continue with next $g
            break;
        }
        self::assertNotNull($parent);
    }
}
```
In general, I agree with @A-Daneel that `self::assertTrue(count($allgroups) > 0)` should be an exception, instead of the rule.

Reply:
I agree that as a unit test it's hard to test more precisely. However testing for the return value being larger then zero is very error prone, you have no idea what you get back. If you could implement @jasperweyne's change it would be nice. Same goes for the `activityRepositoryTest`.

Reply:
Implemented